src/apps/components/AppFrame/AppIFrame.tsx

```diff
@@ -0,0 +1,56 @@
+import { AppDetailsUrlQueryParams, AppUrls } from "@dashboard/apps/urls";
+import { ThemeType } from "@saleor/app-sdk/app-bridge";
+import { useTheme } from "@saleor/macaw-ui";
+import isEqualWith from "lodash/isEqualWith";
+import React, { forwardRef, memo, useEffect, useRef } from "react";
+
+interface AppIFrameProps {
+  appId: string;
+  src: string;
+  featureFlags: Record<string, string>;
+  params: AppDetailsUrlQueryParams;
+  onLoad: () => void;
+  onError: () => void;
+  className: string;
+}
+
+const _AppIFrame = forwardRef<HTMLIFrameElement, AppIFrameProps>(
+  ({ appId, src, featureFlags, params, onLoad, onError, className }, ref) => {
+    const themeRef = useRef<ThemeType>();
+    const { themeType } = useTheme();
+
+    // Ignore updates to themeType - iframe will be notified via events
+    // Otherwise this will cause reload of entire iframe
+    useEffect(() => {
+      themeRef.current = themeType;
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const iframeSrc = AppUrls.resolveAppIframeUrl(appId, src, {
+      ...params,
+      featureFlags,
+      theme: themeRef.current,
```

Comment:
**question**: if we rely on events do you need to pass the theme here?

Reply:
@krzysztofzuraw yes, app can read theme before AppBridge is mounted and set theme early